projects/RabbitMQ.Client/client/impl/ModelBase.cs

```diff
@@ -66,19 +65,11 @@ internal abstract class ModelBase : IFullModel, IRecoverable
 
         public IConsumerDispatcher ConsumerDispatcher { get; }
 
-        public ModelBase(ISession session) : this(session, session.Connection.ConsumerWorkService)
-        { }
-
-        public ModelBase(ISession session, ConsumerWorkService workService)
+        protected ModelBase(bool dispatchAsync, int concurrency, ISession session)
         {
-            if (workService is AsyncConsumerWorkService asyncConsumerWorkService)
-            {
-                ConsumerDispatcher = new AsyncConsumerDispatcher(this, asyncConsumerWorkService);
-            }
-            else
-            {
-                ConsumerDispatcher = new ConcurrentConsumerDispatcher(this, workService);
-            }
+            ConsumerDispatcher = dispatchAsync ?
```

Comment:
Each model now has it's own dispatcher. (Before Dispatcher own, work service shared and workpool own) => This means we could technically expose different dispatcher for different channels! @michaelklishin Is this something we'd want? Do you see an actual usecase for this?

Reply:
I don't. Originally the point of having a dispatcher was to guarantee per-channel ordering of dispatched operations. Now it sort of becomes a part of the channel, just as a separate abstraction. Which is fine with me.

Reply:
Then let's not do it until someone requests it with a valid use case. (Also, you can already do this if you use different connections, so workaround exist already anyway.